src/lib/conf/conftypes.h

```diff
@@ -110,6 +110,16 @@ typedef struct struct_magic_decl_t {
  * fetch this option should produce a warning.
```

Comment:
It's weird having obsolete / no dump / invisible options, and then flags that do similar things a higher level of abstraction:
* contained = no dump, no copy, no check
* not settable = no dump?

And then we have obsolete at this level, which means contained.

I feel like there is a better design that has orthogonal flags, at least at the lowest level. And then the higher levels can use combinations of those flags for contained, settable, obsolete, etc.

Reply:
Agreed. Blocker for this PR, or blocker for calling Tor#29211 done?

Reply:
Must solve before we call #29211 finished, otherwise this code will be very hard to modify in future.